va/va.h

```diff
@@ -4620,6 +4620,46 @@ typedef struct _VAPictureHEVC
  */
 #define VA_PICTURE_HEVC_RPS_LT_CURR             0x00000040
 
+typedef enum{
+    VACopyObjectSurface = 0,
+    VACopyObjectBuffer  = 1,
+    VACopyObjectUserPTR = 2
+} VACopyObjectType;
+
+typedef struct _VACopyObject {
+    VACopyObjectType  obj_type;    // type of object.
+    union
+    {
+        VASurfaceID surface_id;
+        VABufferID  buffer_id;
+        uint8_t *   userptr;       // pointer to system memory
+    } object;
+
+    uint32_t    size;              // for userptr, it should be same with surface/buffer size
+    uint32_t    width_stride;      // for user ptr only,
+    uint32_t    height_stride;     // for user ptr only
+    uint32_t    va_reserved[VA_PADDING_LOW];
+} VACopyObject;
+
+#define VA_COPY_BLOCK    0x00000001
+#define VA_COPY_NONBLOCK 0x00000002
+
+/** \brief Copies an object.
+ *
+ * Copies specified object (surface or buffer). If non-blocking copy
+ * is requested (VA_COPY_NONBLOCK), sync_handle will contain a handle
+ * to pass to vaCopyWait to wait for completion.
+ *
+ * @param[in] dpy               the VA display
+ * @param[in] src               Source object to copy from
+ * @param[in,out] buf_info      Destination object to copy to
+ * @param[out] sync_handle      Synchronization handle to sync on
+ * @return VA_STATUS_SUCCESS if successful
+ */
+VAStatus vaCopy(VADisplay dpy, VACopyObject * src, VACopyObject * dst, uint32_t flags, uintptr_t* sync_handle);
```

Comment:
A proposal: within dGFX should we consider cross adapter (or at least cross vaDisplay) copy? For now it’s not the case, but might become needed later. But later changes are always hard. 
So probably VACopyObject should include VADisplay for VASurfaceID/VABufferID.

Reply:
Updated here: https://github.com/intel/libva/pull/243#issuecomment-672289401